gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -367,13 +412,17 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
                         kspTask.compilerPluginClasspath = project.configurations.getByName(pluginConfigurationName)
                         kspTask.commonSources.from(kotlinCompileTask.commonSources)
                         kspTask.compilerPluginOptions.addPluginArgument(kotlinCompileTask.compilerPluginOptions)
+                        val kspOptions = kspTask.options.get().flatMap { listOf("-P", it.toArg()) }
+                        kspTask.compilerOptions.freeCompilerArgs.value(
+                            kspOptions + kotlinCompileTask.compilerOptions.freeCompilerArgs.get()
+                        )
```

Comment:
If `additionalCompilerOptions` was removed, are there any alternatives? Ideally, these are best set at execution time.

Reply:
the deprecation message says `Replaced with compilerOptions.freeCompilerArgs`